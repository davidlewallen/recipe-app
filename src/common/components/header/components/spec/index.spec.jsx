import React from 'react';
import { shallow } from 'enzyme';

import Header from '../';

describe('Header component', () => {
  it('should match snapshot', () => {
    const component = shallow(<Header logout={jest.fn()} />);
    expect(component).toMatchSnapshot();
  });
});
