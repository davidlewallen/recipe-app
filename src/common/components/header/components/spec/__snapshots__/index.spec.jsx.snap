// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    logout={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsStyle": "default",
      "children": Array [
        <NavbarHeader>
          <NavbarBrand>
            <Link
              replace={false}
              to="/dashboard"
            >
              My Saved Recipes
            </Link>
          </NavbarBrand>
          <NavbarToggle />
        </NavbarHeader>,
        <NavbarCollapse>
          <Nav
            bsClass="nav"
            justified={false}
            pullLeft={false}
            pullRight={true}
            stacked={false}
          >
            <NavItem
              active={false}
              disabled={false}
              eventKey={1}
              href="#"
            >
              + Add Recipe
            </NavItem>
            <LinkContainer
              activeClassName="active"
              exact={false}
              replace={false}
              strict={false}
              to="/account"
            >
              <NavItem
                active={false}
                disabled={false}
                eventKey={2}
              >
                Account
              </NavItem>
            </LinkContainer>
            <NavItem
              active={false}
              disabled={false}
              eventKey={3}
              onClick={[MockFunction]}
            >
              Logout
            </NavItem>
          </Nav>
        </NavbarCollapse>,
      ],
      "fluid": true,
      "inverse": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavbarBrand>
              <Link
                replace={false}
                to="/dashboard"
              >
                My Saved Recipes
              </Link>
            </NavbarBrand>,
            <NavbarToggle />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Link
                replace={false}
                to="/dashboard"
              >
                My Saved Recipes
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "My Saved Recipes",
                "replace": false,
                "to": "/dashboard",
              },
              "ref": null,
              "rendered": "My Saved Recipes",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Nav
            bsClass="nav"
            justified={false}
            pullLeft={false}
            pullRight={true}
            stacked={false}
          >
            <NavItem
              active={false}
              disabled={false}
              eventKey={1}
              href="#"
            >
              + Add Recipe
            </NavItem>
            <LinkContainer
              activeClassName="active"
              exact={false}
              replace={false}
              strict={false}
              to="/account"
            >
              <NavItem
                active={false}
                disabled={false}
                eventKey={2}
              >
                Account
              </NavItem>
            </LinkContainer>
            <NavItem
              active={false}
              disabled={false}
              eventKey={3}
              onClick={[MockFunction]}
            >
              Logout
            </NavItem>
          </Nav>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "nav",
            "children": Array [
              <NavItem
                active={false}
                disabled={false}
                eventKey={1}
                href="#"
              >
                + Add Recipe
              </NavItem>,
              <LinkContainer
                activeClassName="active"
                exact={false}
                replace={false}
                strict={false}
                to="/account"
              >
                <NavItem
                  active={false}
                  disabled={false}
                  eventKey={2}
                >
                  Account
                </NavItem>
              </LinkContainer>,
              <NavItem
                active={false}
                disabled={false}
                eventKey={3}
                onClick={[MockFunction]}
              >
                Logout
              </NavItem>,
            ],
            "justified": false,
            "pullLeft": false,
            "pullRight": true,
            "stacked": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": "+ Add Recipe",
                "disabled": false,
                "eventKey": 1,
                "href": "#",
              },
              "ref": null,
              "rendered": "+ Add Recipe",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "children": <NavItem
                  active={false}
                  disabled={false}
                  eventKey={2}
                >
                  Account
                </NavItem>,
                "exact": false,
                "replace": false,
                "strict": false,
                "to": "/account",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": "Account",
                  "disabled": false,
                  "eventKey": 2,
                },
                "ref": null,
                "rendered": "Account",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": "Logout",
                "disabled": false,
                "eventKey": 3,
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Logout",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "default",
        "children": Array [
          <NavbarHeader>
            <NavbarBrand>
              <Link
                replace={false}
                to="/dashboard"
              >
                My Saved Recipes
              </Link>
            </NavbarBrand>
            <NavbarToggle />
          </NavbarHeader>,
          <NavbarCollapse>
            <Nav
              bsClass="nav"
              justified={false}
              pullLeft={false}
              pullRight={true}
              stacked={false}
            >
              <NavItem
                active={false}
                disabled={false}
                eventKey={1}
                href="#"
              >
                + Add Recipe
              </NavItem>
              <LinkContainer
                activeClassName="active"
                exact={false}
                replace={false}
                strict={false}
                to="/account"
              >
                <NavItem
                  active={false}
                  disabled={false}
                  eventKey={2}
                >
                  Account
                </NavItem>
              </LinkContainer>
              <NavItem
                active={false}
                disabled={false}
                eventKey={3}
                onClick={[MockFunction]}
              >
                Logout
              </NavItem>
            </Nav>
          </NavbarCollapse>,
        ],
        "fluid": true,
        "inverse": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavbarBrand>
                <Link
                  replace={false}
                  to="/dashboard"
                >
                  My Saved Recipes
                </Link>
              </NavbarBrand>,
              <NavbarToggle />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/dashboard"
                >
                  My Saved Recipes
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "My Saved Recipes",
                  "replace": false,
                  "to": "/dashboard",
                },
                "ref": null,
                "rendered": "My Saved Recipes",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Nav
              bsClass="nav"
              justified={false}
              pullLeft={false}
              pullRight={true}
              stacked={false}
            >
              <NavItem
                active={false}
                disabled={false}
                eventKey={1}
                href="#"
              >
                + Add Recipe
              </NavItem>
              <LinkContainer
                activeClassName="active"
                exact={false}
                replace={false}
                strict={false}
                to="/account"
              >
                <NavItem
                  active={false}
                  disabled={false}
                  eventKey={2}
                >
                  Account
                </NavItem>
              </LinkContainer>
              <NavItem
                active={false}
                disabled={false}
                eventKey={3}
                onClick={[MockFunction]}
              >
                Logout
              </NavItem>
            </Nav>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "nav",
              "children": Array [
                <NavItem
                  active={false}
                  disabled={false}
                  eventKey={1}
                  href="#"
                >
                  + Add Recipe
                </NavItem>,
                <LinkContainer
                  activeClassName="active"
                  exact={false}
                  replace={false}
                  strict={false}
                  to="/account"
                >
                  <NavItem
                    active={false}
                    disabled={false}
                    eventKey={2}
                  >
                    Account
                  </NavItem>
                </LinkContainer>,
                <NavItem
                  active={false}
                  disabled={false}
                  eventKey={3}
                  onClick={[MockFunction]}
                >
                  Logout
                </NavItem>,
              ],
              "justified": false,
              "pullLeft": false,
              "pullRight": true,
              "stacked": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": "+ Add Recipe",
                  "disabled": false,
                  "eventKey": 1,
                  "href": "#",
                },
                "ref": null,
                "rendered": "+ Add Recipe",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeClassName": "active",
                  "children": <NavItem
                    active={false}
                    disabled={false}
                    eventKey={2}
                  >
                    Account
                  </NavItem>,
                  "exact": false,
                  "replace": false,
                  "strict": false,
                  "to": "/account",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "children": "Account",
                    "disabled": false,
                    "eventKey": 2,
                  },
                  "ref": null,
                  "rendered": "Account",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": "Logout",
                  "disabled": false,
                  "eventKey": 3,
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": "Logout",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
